pkg/commands/options/build.go

```diff
@@ -67,6 +68,12 @@ type BuildOptions struct {
 
 	// BuildConfigs stores the per-image build config from `.ko.yaml`.
 	BuildConfigs map[string]build.Config
+
+	// ImageConfigs stores the per-image image config from `.ko.yaml`.
+	ImageConfigs map[string]build.ImageConfig `yaml:",omitempty"`
+
+	// Envs stores the global environment variables from `.ko.yaml`.
+	Envs map[string]string `yaml:",omitempty"`
```

Comment:
(grr GitHub ate my comment here, twice!)

This might be confusing because it kinda sounds like these envs will end up getting set in the resulting image(s), like `builds[id].env` does today.

At the same time, we've only gotten the request so far to set `KO_DOCKER_REPO` if the env is unset. Maybe just make this `dockerRepo string` and use it if it's set and `KO_DOCKER_REPO` isn't.

(We'll also need to update docs, and the error message when `KO_DOCKER_REPO` is not set)

Reply:
okay, makes sense to me, I've updated the code according to your comments 🙋🏻‍♂️